File: app-android/src/main/java/com/ustadmobile/port/android/view/LocationLayoutType.java
package com.ustadmobile.port.android.view;

import com.toughra.ustadmobile.R;

import tellh.com.recyclertreeview_lib.LayoutItemType;

public class LocationLayoutType implements LayoutItemType {
    public String name;
    public long uid;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public long getUid() {
        return uid;
    }

    public void setUid(long uid) {
        this.uid = uid;
    }

    public LocationLayoutType(String name) {
        this.name = name;
    }

    public LocationLayoutType(String name, Long uid){
        this.name = name;
        this.uid = uid;
    }

    @Override
    public int getLayoutId() {
        return R.layout.item_select_multiple_tree_dialog;
    }
}

File: app-android/src/main/kotlin/com/ustadmobile/staging/port/android/view/LocationLayoutType.kt
package com.ustadmobile.staging.port.android.view

import com.toughra.ustadmobile.R

import tellh.com.recyclertreeview_lib.LayoutItemType

class LocationLayoutType : LayoutItemType {
    var name: String
    var uid: Long = 0
    var selected: Boolean = false
    var leaf: Boolean = false

    constructor(name: String) {
        this.name = name
    }

    constructor(name: String, uid: Long?, selected: Boolean, leaf: Boolean) {
        this.name = name
        this.uid = uid!!
        this.selected = selected
        this.leaf = leaf
    }

    override fun getLayoutId(): Int {
        return R.layout.item_select_multiple_tree_dialog
    }
}
